src/lib/viewers/controls/media/MediaSettingsControls/MediaSettingsFlyout.tsx

```diff
@@ -0,0 +1,69 @@
+import React from 'react';
+import classNames from 'classnames';
+import { Menu } from './MediaSettingsMenu';
+import './MediaSettingsFlyout.scss';
+
+export type Props = {
+    children: React.ReactNode;
+    className?: string;
+    menu: Menu;
+    menuSelectors: Record<string, string>;
+};
+
+const SETTINGS_MENU_PADDING = 18;
+const SETTINGS_MENU_PADDING_SCROLL = 32;
+const SETTINGS_MENU_MAX_HEIGHT = 210;
+
+export default function MediaSettingsFlyout({ children, className, menu, menuSelectors }: Props): JSX.Element | null {
+    const [dimension, setDimension] = React.useState({});
+    const flyoutRef = React.useRef<HTMLDivElement>(null);
+
+    const handleTransitionEnd = (): void => {
+        const { current: flyoutEl } = flyoutRef;
+
+        if (flyoutEl) {
+            flyoutEl.classList.remove('bp-MediaSettingsFlyout-in-transition');
+        }
+    };
+
+    React.useEffect(() => {
+        const { current: flyoutEl } = flyoutRef;
+
+        if (flyoutEl) {
+            flyoutEl.addEventListener('transitionend', handleTransitionEnd);
+        }
+
+        return (): void => {
+            if (flyoutEl) {
+                flyoutEl.removeEventListener('transitionend', handleTransitionEnd);
+            }
+        };
+    }, [flyoutRef]);
+
+    React.useEffect(() => {
+        const nextMenuEl = document.querySelector(menuSelectors[menu]);
+        const { current: flyoutEl } = flyoutRef;
+
+        if (nextMenuEl && flyoutEl) {
+            flyoutEl.classList.add('bp-MediaSettingsFlyout-in-transition');
+
+            const { width, height } = nextMenuEl.getBoundingClientRect();
+            const paddedHeight = height + SETTINGS_MENU_PADDING;
+
+            // If the menu grows tall enough to require scrolling, take into account scroll bar width.
+            const scrollPadding =
+                paddedHeight >= SETTINGS_MENU_MAX_HEIGHT ? SETTINGS_MENU_PADDING_SCROLL : SETTINGS_MENU_PADDING;
```

Comment:
Would using `minHeight` and `minWidth` allow us to keep the padding values in the stylesheet, instead of including them in the calculated value?

Reply:
I copied the constants and this formula from original code. Maybe we can make it better and in another PR.